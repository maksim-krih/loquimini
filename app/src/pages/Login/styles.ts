import {createUseStyles} from 'react-jss'

export const useStyles = createUseStyles({
  container: {
    backgroundColor: "#001529",
    position: "fixed",
    width: "100%",
    height: "100%"
  },
  login: {
    padding: "0 24px",
    width: 300,
    margin: "auto",
    marginTop: 300
  }
})