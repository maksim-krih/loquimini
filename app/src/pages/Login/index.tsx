import React, { FC, useState } from "react";
import { Button, Input } from "antd";
import { IProps } from "./types";
import { useStyles } from "./styles";
import Api from "../../services";

const { Password } = Input;

const Login: FC<IProps> = (props: IProps) => {
  const classes = useStyles();

  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");

  const onLogin = () => {
    Api.Auth.login({
      email,
      password
    });
  };

  return (
    <div className={classes.container}>
      <Input value={email} onChange={(e) => setEmail(e.target.value)} />
      <Password value={password} onChange={(e) => setPassword(e.target.value)} />
      <Button onClick={onLogin}>Sign in</Button>
    </div>
  );
}

export default Login;