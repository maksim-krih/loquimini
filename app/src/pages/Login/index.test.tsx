import Login from ".";
import { shallow } from "enzyme";

const setUp = () => shallow(<Login />);

describe("Login", () => {
  it("should render Login Component", () => {
    const component = setUp();
    expect(component).toMatchSnapshot();
  });
});