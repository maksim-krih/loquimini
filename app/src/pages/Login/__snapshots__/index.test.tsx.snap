// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render Login Component 1`] = `
<div
  className="container-0-2-1"
>
  <div
    className="login-0-2-2"
  >
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "justifyContent": "center",
          "marginBottom": 10,
        }
      }
    >
      <img
        alt="logo"
        src="logo.png"
        style={
          Object {
            "marginRight": 10,
            "width": 48,
          }
        }
      />
      <Title
        level={3}
        style={
          Object {
            "color": "#001529",
          }
        }
      >
        Loquimini
      </Title>
    </div>
    <ForwardRef(InternalForm)
      form={
        Object {
          "__INTERNAL__": Object {
            "itemRef": [Function],
          },
          "getFieldError": [Function],
          "getFieldInstance": [Function],
          "getFieldValue": [Function],
          "getFieldsError": [Function],
          "getFieldsValue": [Function],
          "getInternalHooks": [Function],
          "isFieldTouched": [Function],
          "isFieldValidating": [Function],
          "isFieldsTouched": [Function],
          "isFieldsValidating": [Function],
          "resetFields": [Function],
          "scrollToField": [Function],
          "setFields": [Function],
          "setFieldsValue": [Function],
          "submit": [Function],
          "validateFields": [Function],
        }
      }
      labelCol={
        Object {
          "span": 6,
        }
      }
      name="login"
      onFinish={[Function]}
      onFinishFailed={[Function]}
      validateMessages={
        Object {
          "required": "\${label} is required!",
        }
      }
      wrapperCol={
        Object {
          "span": 16,
        }
      }
    >
      <FormItem
        label="Email"
        name="email"
        rules={
          Array [
            Object {
              "required": true,
            },
          ]
        }
      >
        <Input
          style={
            Object {
              "borderRadius": 5,
            }
          }
          type="text"
        />
      </FormItem>
      <FormItem
        label="Password"
        name="password"
        rules={
          Array [
            Object {
              "required": true,
            },
          ]
        }
      >
        <Password
          action="click"
          iconRender={[Function]}
          style={
            Object {
              "borderRadius": 5,
            }
          }
          visibilityToggle={true}
        />
      </FormItem>
    </ForwardRef(InternalForm)>
    <div
      style={
        Object {
          "marginBottom": 10,
        }
      }
    />
    <Button
      loading={false}
      onClick={[Function]}
      style={
        Object {
          "width": "100%",
        }
      }
      type="primary"
    >
      Sign in
    </Button>
  </div>
</div>
`;
